docs/pages/core-concepts.mdx

```diff
@@ -0,0 +1,219 @@
+---
+title: Teleport Core Concepts
+description: Learn the the key architectural components that make up Teleport.
+---
+
+Here are the core concepts that make up a Teleport deployment. You will see
+these terms often when setting up and managing Teleport, so you should get
+familiar with them before following other pages in the documentation.
+
+## Authentication connector
+
+An authentication connector is a **configuration resource** that allows users to
+authenticate to Teleport via a Single Sign-On (SSO) solution. 
+
+See our guide to [Authentication Options](./reference/authentication.mdx).
+
+## Certificate authority
+
+Teleport allows services and clients to achieve secure access using SSH and TLS
+certificates. 
+
+The components of a **Teleport Cluster**, such as clients and **Teleport
+Services**, use these certificates to verify that they are connecting with a
+genuine Teleport Service or user. To do so, they check that the certificate was
+signed by a **certificate authority**—an entity that issues certificates—that
+belongs to your Teleport Cluster. 
+
+The **Teleport Auth Service** includes multiple certificate authorities, which
+it uses for different purposes.
+
+Read about how Teleport's certificate authorities work in our [Auth Service
+architecture guide](./architecture/authentication.mdx).
+
+## Configuration resource
+
+A **configuration resource** is a document stored on the **Auth Service**
+backend that specifies settings for your **Teleport cluster**. Examples include
+**roles**, **local users**, and **authentication connectors**
+
+Read more in our [Resource Reference](./reference/resources.mdx).
+
+## Machine ID
```

Comment:
It's worth mentioning here that Machine ID is commonly used for machine-to-machine and service-to-service use cases.